src/azure_pdf_parser/utils.py

```diff
@@ -29,74 +32,125 @@ def call_api_with_error_handling(retries: int, func, *args, **kwargs) -> Any:
                 raise e
 
 
-def propagate_page_number(page: PDFPage) -> PDFPage:
-    """Propagate the page number to the paragraphs and tables."""
-    if page.extracted_content.paragraphs:
-        for paragraph in page.extracted_content.paragraphs:
+def propagate_page_number(batch: PDFPagesBatchExtracted) -> PDFPagesBatchExtracted:
+    """
+    Correct the page numbers in the batch.
+
+    This is done by propagating the page number of the start index of the batch to the
+    paragraphs and tables.
+
+    Page number in the batch is incremented as follows:
+
+    [1,2,3,4] and a page range of 101-104 -> [101,102,103,104]
+    Thus, incremented page number = page number + batch.page_range[0] - 1
+
+    E.g.
+    - page number 1 in the batch is page number 101 in the document (1 + 101 - 1).
+    - page number 2 in the batch is page number 102 in the document (2 + 101 - 1).
+    """
+    if batch.extracted_content.paragraphs:
```

Comment:
I think it might be worth pulling out the page offset, just to make it harder to lose a `-1` somewhere

```suggestion
    page_offset = batch.page_range[0] - 1

    if batch.extracted_content.paragraphs:
```